### PR TITLE
Let the configuration cache report be a self contained file

### PR DESCRIPTION
**Pull request by eskatos**
Before this PR users had to create a zip of all the report files before being able to share it properly.

This PR makes the configuration cache HTML report be generated as a single self contained file.
This makes it very easy to share, upload with bug reports etc..

Additionally, tests are added to prevent breaking the development workflow for iterating on the report code, and the report code quality is now tested.

### CODE REVIEW

File: subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionReport.kt [hunk 1]
```diff
@@ -83,41 +82,52 @@ class InstantExecutionReport(
...
+                writer.writeReportFileText(reader, cacheAction, problems)
+            }
         }
     }
 
+    private
+    fun BufferedWriter.writeReportFileText(htmlReader: BufferedReader, cacheAction: String, problems: List<PropertyProblem>) {
+        var dataWritten = false
+        htmlReader.forEachLine { line ->
+            if (line.contains("configuration-cache-report-data.js")) {
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionReport.kt:108]: New implementation should be much simpler.

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/instantexecution/InstantExecutionProblemsFixture.groovy [hunk 2]
```diff
@@ -306,13 +304,25 @@ final class InstantExecutionProblemsFixture {
...
         // InstantExecutionReport ensures the pure json model can be read
-        // by skipping the 1st and last line of the report data file
-        def jsonText = reportDataFile.readLines().with { subList(1, size() - 1) }.join('\n')
+        // by looking for `// begin-report-data` and `// end-report-data`
+        def jsonText = linesBetween(reportFile, '// begin-report-data', '// end-report-data')
+        assert jsonText: "malformed report file"
         new JsonSlurper().parseText(jsonText) as Map<String, Object>
     }
 
+    private static String linesBetween(File file, String beginLine, String endLine) {
```
Comment (bamboo) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/instantexecution/InstantExecutionProblemsFixture.groovy:315]: 🤩

**Pull request merged**